pkg/controller/association/conf.go

```diff
@@ -103,6 +107,39 @@ func ElasticsearchAuthSettings(c k8s.Client, association commonv1.Association) (
 	return assocConf.AuthSecretKey, string(data), nil
 }
 
+// AllowVersion returns true if the given resourceVersion is lower or equal to the associations' versions.
+// For example: Kibana in version 7.8.0 cannot be deployed if its Elasticsearch association reports version 7.7.0.
+// Referenced resources version is parsed from the association conf annotation.
+func AllowVersion(resourceVersion version.Version, associated commonv1.Associated, logger logr.Logger, recorder record.EventRecorder) bool {
+	for _, assoc := range associated.GetAssociations() {
+		assocRef := assoc.AssociationRef()
+		if !assocRef.IsDefined() {
+			// no association specified, move on
+			continue
+		}
+		if assoc.AssociationConf() == nil || assoc.AssociationConf().Version == "" {
+			// no conf reported yet, this may be the initial resource creation
+			logger.Info("Delaying version deployment since the version of an associated resource is not reported yet",
+				"version", resourceVersion, "ref_namespace", assocRef.Namespace, "ref_name", assocRef.Name)
+			return false
+		}
+		refVer, err := version.Parse(assoc.AssociationConf().Version)
+		if err != nil {
+			logger.Error(err, "Invalid version found in association conf", "association_version", assoc.AssociationConf().Version)
+			return false
+		}
+		if !refVer.IsSameOrAfter(resourceVersion) {
```

Comment:
This is stricter than required, correct? https://www.elastic.co/support/matrix#matrix_compatibility

They're forwards and backwards compatible within the same minor (i.e. Kibana 7.8.1 is compatible with ES 7.8.0), but here we are checking the patch version too.

Reply:
That's a good question. I was mostly following the [stack upgrade process](https://www.elastic.co/guide/en/elastic-stack/current/upgrading-elastic-stack.html#upgrade-order-elastic-stack), but the matrix indeed shows backward compatibility with the same minor (and with the same major in most cases).
I'll try to get more details about the right thing to do.

Reply:
I asked around and we discussed this in a different channel: it makes sense to restrict the upgrade ordering to major and minors only, and relax it for different patch versions. In other terms: allow Kibana v7.8.1 alongside Elasticsearch 7.8.0, but not Kibana 7.9.0 along side Elasticsearch 7.8.1.

fdd6d99 fixes it.